addons/binding/org.openhab.binding.tankerkoenig/src/main/java/org/openhab/binding/tankerkoenig/handler/BridgeHandler.java

```diff
@@ -0,0 +1,512 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.tankerkoenig.handler;
+
+import java.math.BigDecimal;
+import java.text.ParseException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+
+import org.joda.time.DateTime;
+import org.joda.time.LocalDate;
+import org.joda.time.LocalTime;
+import org.openhab.binding.tankerkoenig.TankerkoenigBindingConstants;
+import org.openhab.binding.tankerkoenig.internal.config.LittleStation;
+import org.openhab.binding.tankerkoenig.internal.config.OpeningTime;
+import org.openhab.binding.tankerkoenig.internal.config.OpeningTimes;
+
+import org.openhab.binding.tankerkoenig.internal.config.TankerkoenigListResult;
+import org.openhab.binding.tankerkoenig.internal.data.TankerkoenigService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BridgeHandler} is responsible for handling the things (Tankstellen)
+ *
+ *
+ * @author Dennis Dollinger/Jürgen Baginski
+ */
+public class BridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private String apiKey;
+    private int refreshInterval;
+    private boolean setupMode;
+    private boolean use_OpeningTime;
+
+    private ArrayList<Thing> tankstellenThingList;
+    private HashMap<String, LittleStation> tankstellenList;
+
+    TankerkoenigListResult tankerkoenigListResult;
+
+    private ScheduledFuture<?> pollingJob;
+
+    public BridgeHandler(Bridge bridge) {
+        super(bridge);
+        this.tankstellenThingList = new ArrayList<Thing>();
+        tankstellenList = new HashMap<String, LittleStation>();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no code needed.
+    }
+
+    @Override
+    public void initialize() {
+        Configuration config = getThing().getConfiguration();
+        this.setApiKey((String) config.get(TankerkoenigBindingConstants.CONFIG_API_KEY));
+        this.setRefreshInterval(((BigDecimal) config.get(TankerkoenigBindingConstants.CONFIG_REFRESH)).intValue());
+        this.setSetupMode((boolean) config.get(TankerkoenigBindingConstants.CONFIG_SETUP_MODE));
+        this.setUseOpeningTime((boolean) config.get(TankerkoenigBindingConstants.CONFIG_USE_OPENINGTIME));
+
+        updateStatus(ThingStatus.ONLINE);
```

Comment:
Actually, my binding implemented a simililar approach at first. But then, Kai asked me to set the binding to UNKNOWN (basically your "Setup Mode") at configuration time, and having the Thing be ONLINE or OFFLINE according to the real status of the connection.